lib/alerts/Alerts.ts

```diff
@@ -0,0 +1,63 @@
+import { EventEmitter } from 'events';
+import { BalanceAlert } from './types';
+import SwapClientManager from '../swaps/SwapClientManager';
+import { MIN_BALANCE_ALERT_THRESHOLD_IN_MS } from './consts';
+import Logger from '../Logger';
+import { AlertType, ChannelSide } from '../constants/enums';
+import { satsToCoinsStr } from '../cli/utils';
+
+interface Alerts {
+  on(event: 'alert', listener: (alert: any) => void): this;
+  emit(event: 'alert', alert: any): boolean;
+}
+
+// TODO this class still requires a cleanup if alert is not being thrown anymore after a while
+/**
+ * This class works as a middleware for thrown alerts from xud's main flow. Each alert will be caught here
+ * and re-thrown if last thrown time was before the minimum threshold that set in consts.ts
+ */
+class Alerts extends EventEmitter {
+  private alerts = new Map<string, number>();
+  private logger: Logger;
+
+  constructor({ swapClientManager, logger }: { swapClientManager: SwapClientManager; logger: Logger }) {
+    super();
+    this.logger = logger;
+    this.listenLowTradingBalanceAlerts(swapClientManager);
+  }
+
+  private listenLowTradingBalanceAlerts(swapClientManager: SwapClientManager) {
+    const lndClients = swapClientManager.getLndClientsMap().values();
+    for (const lndClient of lndClients) {
+      lndClient.on('lowTradingBalance', this.onLowTradingBalance);
+    }
+    swapClientManager.connextClient?.on('lowTradingBalance', this.onLowTradingBalance);
+  }
+
+  private onLowTradingBalance = (balanceAlert: BalanceAlert) => {
+    const stringRepresentation = JSON.stringify(balanceAlert);
```

Comment:
I think I see the desire to convert to a string here, but I'm concernedthis will cause some issues using plain old JSON.stringify, which isn't guaranteed to give the same string for equivalent objects. It's also a bit expensive of a procedure, and uses up more memory in the map.

Doesn't have to be addressed in this PR since it's not that simple, but maybe it can be addressed at the same time as cleanup. I'm thinking it'd make sense to add a timestamp property to the `Alert` object, and then convert alert's to a string identifier like `[type]-[channel_id]` rather than converting the entire object to a json string.

For now I think add a TODO here and it's ok with me.

Reply:
Yes, you're right, alerts already have timestamps attached within BaseAlert, adding `TODO` there, but I guess we should prepare **cleanup** as soon as possible to avoid memory issues.

Reply:
Maybe even within this PR, wdyt? @sangaman

Reply:
Up to you, it's not a trivial effort.